docs/quickstarts/cosign-akv-azure.md

```diff
@@ -0,0 +1,441 @@
+    ```
+
+    To learn more about Azure CLI and how to sign in with it, see [Sign in with Azure CLI](https://learn.microsoft.com/en-us/cli/azure/authenticate-azure-cli).
+
+1. Create a key in your AKV
+
+    Ensure the logged-in identity assigned role `Key Vault Crypto Officer` for creating a key in AKV.
+
+    ```bash
+    az keyvault key create --vault-name $AKV_NAME -n $KEY_NAME --protection software
+    ```
+
+    Get the key id and retrieve the key version:
+
+    ```bash
+    az keyvault key show --name $KEY_NAME --vault-name $AKV_NAME --query "key.kid"
+    ```
+
+    An example output:
+
+    ```text
+    https://<your akv name>.vault.azure.net/keys/<your key name>/<version>
+    ```
+
+    Configure an environment variable for the version:
+
+    ```bash
+    export KEY_VER=<version>
+    ```
+
+1. Build two images in your ACR
+
+    Login to ACR and ensure the logged-in users has the role `acrpush` and `acrpull` roles assigned.
+
+    ```bash
+    az acr login --name $ACR_NAME
+    ```
+
+    Build and push an image that will be signed in later steps.
+
+    ```bash
+    az acr build -r $ACR_NAME -t $IMAGE_SIGNED $IMAGE_SIGNED_SOURCE --no-logs
+    ```
+
+    Build and push an image that will not be signed.
+
+    ```bash
+    az acr build -r $ACR_NAME -t $IMAGE_UNSIGNED $IMAGE_UNSIGNED_SOURCE --no-logs
+    ```
+
+1. Sign an image stored in your ACR
+
+    Confirm no signatures before signing
+
+    ```bash
+    cosign tree $IMAGE_SIGNED
+    ```
+    
+    Sign the image
+
+    ```bash
+    cosign sign --key azurekms://$AKV_NAME.vault.azure.net/$KEY_NAME/$KEY_VER --tlog-upload=false $IMAGE_SIGNED
+    ```
+
+    > In this article, use flag `--tlog-upload=false` to skip upload the signature to the transparent log (Rekor by default). 
+    > Sign using key in AKV does not necessarily require the role `Key Vault Crypto Officer`, you can login with another identity and assign the role `Key Vault Crypto User` for signing action.
+
+    Confirm the signature is pushed and associated with the image in ACR
+
+    ```bash
+    cosign tree $IMAGE_SIGNED
+    ```
+
+## Validate signatures at K8s admission control
+
+### Access control
+
+User assigned managed identity is used for Ratify to access ACR and AKV. 
+
+1. Configure environment variables
+
+```bash
+export ID_NAME=<your managed id>
+export ID_RG=<your managed id resource group>
+```
+
+1. Create a user assigned managed identity
+
+    ```bash
+    az identity create --name $ID_NAME --resource-group $ID_RG
+    export OBJECT_ID=$(az identity show --name $ID_NAME --resource-group $ID_RG --query 'principalId' -o tsv)
+    ```
+
+1. Assign `acrpull` role to this identity for accessing ACR
+
+    ```bash
+    az role assignment create --assignee-object-id $OBJECT_ID --role acrpull \
+    --scope "/subscriptions/$SUB_ID/resourceGroups/$ACR_RG/providers/Microsoft.ContainerRegistry/registries/$ACR_NAME"
+    ```
+
+1. Assign `Key Vault Crypto User` role to this identity for accessing AKV
```

Comment:
Just checking here, for keys , is there a security for be able to export  private key?

Reply:
Good questions. I checked the document for the role [Key Vault Crypto User](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/security#key-vault-crypto-user), when read, private key is never exposed. So, users do not need to worry about the role assigned to the identity that Ratify used will read a private key. /cc: @akashsinghal